### PR TITLE
feature/gifting

### PR DESCRIPTION
**Pull request by JamesMBligh**
- Added gift card program list API
- Added get product detail
- Add the gifting quote end point Add the gifting order end point
- Distribute 1.0.6

### CODE REVIEW

File: src/wpay-1.json [hunk 1]
```diff
@@ -10450,6 +10450,767 @@
...
+                  },
+                  "discountAmount": {
+                    "description":"Eligible discount amount. In case of no discounts, value will be 0",
+                    "type":"number"
+                  },
+                  "totalOrderAmount": {
+                    "description":"Net amount payable",
+                    "type":"number"
+                  },
+                  "billingAddress": {
```
Comment (jameskeating87) [inline src/wpay-1.json:11049]: @JamesMBligh - A schema already exists for billing address `#/components/schemas/billingAddress` should it be reused here? The fields looke to be the same although the gifting version lists more of the the fields as required.

**Review by jameskeating87 — APPROVED** [review]

File: src/wpay-1.json (hunk 1 above)
Comment (kierans) [reply src/wpay-1.json:11049]: I agree. This duplication causes problems for the SDKs.

File: src/wpay-1.json [hunk 2]
```diff
@@ -10450,6 +10450,767 @@
...
+              "type":"string"
+            },
+            "in":"header",
+            "required":true,
+            "example":"eyJhbGciOiJSUzI1NiIsImtpZCI6ImRldiIsInZlcmIiOiJQT1NUIiwidXJsIjoiaHR0cHM6Ly9kZXYubW9iaWxlLWFwaS53b29sd29ydGhzLmNvbS5hdS93b3cvdjEvandzZGVtby92YWxpZGF0ZSIsInRpbWVzdGFtcCI6MTU5NTIwNjcxNDQzOH0..muEr0b3GNORrP0FW1ohUh2XITRNaOO7uBz"
+          },
+          {
+            "in":"query",
+            "name":"last-update-date-time",
+            "description":"If present, the date/time to limit products returned.  Only products changed since this time will not be returned",
```
Comment (JamesMBligh) [inline src/wpay-1.json:10512]: Remove the "not"

File: src/wpay-1.json [hunk 3]
```diff
@@ -22154,6 +23174,195 @@
...
+              "required":[
+                "data",
+                "meta"
+              ],
+              "properties":{
+                "data":{
+                  "description":"Mandatory data object containing response",
+                  "type":"object",
+                  "required":[
+                    "transactions"
```
Comment (JamesMBligh) [inline src/wpay-1.json:23193]: "transactions" should be "giftingProducts"

Comment (JamesMBligh) [reply src/wpay-1.json:23193]: Actually, should only be the list reference

File: src/wpay-1.json [hunk 4]
```diff
@@ -22154,6 +23174,195 @@
...
+                        "quantity",
+                        "isGifting"
+                      ],
+                      "type":"object",
+                      "properties":{
+                        "designId": {
+                          "description":"Unique identifier of the design selected (assumed to be DIGITAL only currently)",
+                          "type":"string"
+                        },
+                        "amount": {
```
Comment (JamesMBligh) [inline src/wpay-1.json:23289]: Align price field names to PaymentRequest objects

File: src/wpay-1.json [hunk 5]
```diff
@@ -10450,6 +10450,767 @@
...
+                  "billingAddress",
+                  "orderItems"
+                ],
+                "type":"object",
+                "properties": {
+                  "instrumentId": {
+                    "description": "The instrumentId to be used for the order.  Must not be a stored gift card",
+                    "type": "string"
+                  },
+                  "deliveryEmail": {
```
Comment (JamesMBligh) [inline src/wpay-1.json:11029]: Either clarify field name or make description clearer

**Pull request merged**